packages/rspack-dev-client/src/reactRefresh.ts

```diff
@@ -1,22 +1,26 @@
+// Thanks https://github.com/pmmmwh/react-refresh-webpack-plugin
 // @ts-ignore
+const RefreshUtils = require('@pmmmwh/react-refresh-webpack-plugin/lib/runtime/RefreshUtils');
 const RefreshRuntime = require("react-refresh/runtime");
 
-function debounce(fn: Function, delay: number) {
-	var handle: number | undefined;
-	return () => {
-		clearTimeout(handle);
-		handle = setTimeout(fn, delay);
-	};
-}
-
 RefreshRuntime.injectIntoGlobalHook(globalThis);
-globalThis.$RefreshReg$ = () => {};
-globalThis.$RefreshSig$ = () => type => type;
 
-var queueUpdate = debounce(RefreshRuntime.performReactRefresh, 16);
+// Port from https://github.com/pmmmwh/react-refresh-webpack-plugin/blob/main/loader/utils/getRefreshModuleRuntime.js#L29
+function refresh(moduleId, webpackHot) {
+	const currentExports = RefreshUtils.getModuleExports(moduleId);
+	const fn = () => {
```

Comment:
It seems the function should receive the exported value:
<img width="421" alt="image" src="https://user-images.githubusercontent.com/10465670/229706996-ab9bea10-d304-43a9-b2ac-988aa3264ed5.png">